File: leetcode/3248-snake-in-matrix/finalPositionOfSnake.test.ts
import cases from "./cases.json";
import finalPositionOfSnake from "./finalPositionOfSnake";

test("snake in matrix", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const result = finalPositionOfSnake(
      testCase.input.n,
      testCase.input.commands
    );
    console.timeEnd("time");
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/3248-snake-in-matrix/finalPositionOfSnake_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type FinalPositionOfSnakeCase struct {
	Input struct {
		N        int
		Commands []string
	}

	Output int
}

func TestFinalPositionOfSnake(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]FinalPositionOfSnakeCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := finalPositionOfSnake(testCase.Input.N, testCase.Input.Commands)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
